test_matching.py

```diff
@@ -10,16 +10,12 @@
 class KnockServerTestCase(unittest.TestCase):
 
     def setUp(self):
-        self.db_fd, app.config['DATABASE'] = tempfile.mkstemp()
-        app.config['TESTING'] = True
         self.app = app.test_client()
-        with app.app_context():
-            database.init_db()
+        database.init_db()
```

Comment:
From my testing we still need to manually init the db on setup for the tests to pass. *shrug* can't hurt to leave this.